Gems/Atom/Feature/Common/Assets/Shaders/SkyAtmosphere/SkyVolumeLUT.azsl

```diff
@@ -0,0 +1,102 @@
+
+    float slice = (float(position.z) + 0.5f) / SKY_VOLUME_SLICE_COUNT;
+    slice *= slice;    // squared distribution
+    slice *= SKY_VOLUME_SLICE_COUNT;
+
+    float tMax = AerialPerspectiveSliceToDepth(slice);
+
+    // get position relative to planet surface
+    float3 froxelPosition = cameraPlanetPosKm + tMax * worldDir;
+
+    // if the froxel is under the ground, make sure to offset it out on the ground.
+    float viewHeight = length(froxelPosition);
+    if (viewHeight <= (Atmosphere.BottomRadius + PLANET_RADIUS_OFFSET_KM))
+    {
+        // Apply a position offset to make sure no artefact are visible close to the earth boundaries for large voxel.
+        froxelPosition = normalize(froxelPosition) * (Atmosphere.BottomRadius + PLANET_RADIUS_OFFSET_KM + 0.001f);
+        worldDir = normalize(froxelPosition - cameraPlanetPosKm);
+        tMax = length(froxelPosition - cameraPlanetPosKm);
+    }
+    float maxTraceDepth = tMax;
+
+    const float depth = -1.0;
+
+    // move ray marching start up to top atmosphere.
+    viewHeight = length(froxelPosition);
+    if (viewHeight >= Atmosphere.TopRadius)
+    {
+        float distanceToAtmosphere = 0.f;
+
+        if (!MoveToAtmosphere(Atmosphere, cameraPlanetPosKm, distanceToAtmosphere, worldDir, depth))
+        {
+            // ray is not intersecting the atmosphere
+            return float4(0.0, 0.0, 0.0, 1.0);
+        }
+
+        if (maxTraceDepth < distanceToAtmosphere)
+        {
+            // maxTraceDepth for this voxel is not within earth atmosphere
+            return float4(0.0, 0.0, 0.0, 1.0);
+        }
+
+        // world position has been moved to the atmosphere boundary: we need to reduce maxTraceDepth accordingly. 
+        maxTraceDepth = max(0.0, maxTraceDepth - distanceToAtmosphere);
+    }
+
+    const bool ground = false;
+
+    // setting variableSampleCount false may improve performance, but atmosphere will look thinner 
+    // so you may need to increase sampleCountIni
+    const float sampleCountIni = max(1.0, float(position.z + 1.0) * 2.0f);
+    const bool variableSampleCount = true;
+    const bool mieRayPhase = true;
+    const float2 screenCoords = float2(0,0); // unused
+    SingleScatteringResult ss = IntegrateScatteredLuminance(screenCoords, cameraPlanetPosKm, worldDir, PassSrg::m_constants.m_sunDirection, Atmosphere, ground, sampleCountIni, depth, variableSampleCount, mieRayPhase, maxTraceDepth);
+
+    const float transmittance = dot(ss.Transmittance, float3(1.0f / 3.0f, 1.0f / 3.0f, 1.0f / 3.0f));
+    return float4(ss.L, 1.0 - transmittance);
+}
+
+// total thread count is based on the "Target Thread Count X/Y/Z" in the SkyVolumeLUT.pass file
+// and the number of threads per group specified in numthreads() below.
+// given a volume texture size is 32x32x32, with 1 thread per froxel
+// the following values are based on intention to improve locality when accessing 
+// the transmission LUT
+[numthreads(1, 8, 32)]
```

Comment:
Normally, I would expect a volume texture to be arranged in memory by varying across slices first, and then between slices next. I'm curious if you compared timings by any chance between this manner of thread assignment as opposed to just a simple `numthreads(32, 1, 1)` with each group just emitting its own slice.

Reply:
Any experiments done with different configurations as mentioned by Jeremy above?

Reply:
Not yet, but I intend to try it as soon as I get some time to work on this again.

Reply:
I did some captures of `1,8,32` vs `32,1,1` and the timings for the SkyVolumeLUT pass were;

| Thread Config | Pass Duration | Warp usage | 
|-|-|-|
| 1,8,32| .022ms - .025ms |  100% then taper |
| 32,1,1 | .023ms - .027ms | Medium usage throughout | 

Here's the warp graphs of the two configs:
*1,8,32*
![SkyVolumeLUTPix1_8_32](https://user-images.githubusercontent.com/26804013/224811989-cb79c531-93b8-43b8-9999-a3ba63d542f1.PNG)


*32,1,1*
![SkyVolumeLUTPix32_1_1](https://user-images.githubusercontent.com/26804013/224812004-6f3a2ab4-d750-4390-8952-76f9a68c7cf3.PNG)

Reply:
I believe it depends on the hw. Modern desktop hw is very forgiving. I'd be curious to see how it did on iOS where workgroups of 128 or greater are recommended even though the hw has a native width of 32.

Reply:
Looks like 1,8,32 has better throughput